### PR TITLE
Add dhcp-relay structured events tests (#13504)

### PR DESCRIPTION
**Pull request by zbud-msft**
Cherry-pick #13504 

telemetry/test_events.py::test_events[vlab-01-False] PASSED              [ 33%]
telemetry/test_events.py::test_events_cache[vlab-01-False] PASSED        [ 66%]
telemetry/test_events.py::test_events_cache_overflow[vlab-01-False] PASSED [100%]

Test Results:

22/10/2024 00:44:06 run_events_test.run_test                 L0027 INFO   | events received:
```
([
    {
        "sonic-events-dhcp-relay:dhcp-relay-discard": {
            "ifname": "eth4",
            "timestamp": "2024-10-22T00:43:56.334060Z"
        }
    }
])
```
  22/10/2024 00:47:47 run_events_test.run_test                 L0027 INFO   | events received:
 ```
([
      {
          "sonic-events-dhcp-relay:dhcp-relay-disparity": {
              "duration": "198",
              "timestamp": "2024-10-22T00:47:12.924422Z",
              "vlan": "Agg-Vlan1000"
          }
    }
])
```
22/10/2024 00:48:26 run_events_test.run_test                 L0027 INFO   | events received: 
```
([
    {
        "sonic-events-dhcp-relay:dhcp-relay-bind-failure": {
            "timestamp": "2024-10-22T00:48:25.445207Z",
            "type": "local",
            "vlan": "Vlan1000"
        }
    }
])
```

Add support for dhcp-relay-discard event, which we use ptfadapter to and ptf.testutils to send a dhcp discover packet to a port that has no ipv4 address assigned.

Add support for dhcp-relay-disparity event which uses same repro for dhcp-relay-disparity but at greater intervals and packets sent due to dhcpmon configuration

Added new testcase that triggered sending dhcp discover packet which will trigger event.

Pipeline/Manual test

run_events_test.run_test                 L0027 INFO   | events received:
```json
    {
        "sonic-events-dhcp-relay:dhcp-relay-discard": {
            "ifname": "Vlan1000",
            "timestamp": "2024-07-09T01:27:30.395216Z"
        }
    }
```
run_events_test.run_test                 L0027 INFO   | events received:
```json
    {
        "sonic-events-dhcp-relay:dhcp-relay-disparity": {
            "duration": "198",
            "timestamp": "2024-07-09T01:30:43.711312Z",
            "vlan": "Agg-Vlan1000"
        }
    }
```
run_events_test.run_test                 L0027 INFO   | events received:
```json
    {
        "sonic-events-dhcp-relay:dhcp-relay-bind-failure": {
            "timestamp": "2024-07-09T01:32:27.667714Z",
            "type": "local",
            "vlan": "Vlan1000"
        }
    }
```

### Description of PR

Summary:
Fixes # (issue) 26636373

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [x] 202405

#### What is the motivation for this PR?

Add support for dhcp-relay-discard event, which we use ptfadapter to and ptf.testutils to send a dhcp discover packet to a port that has no ipv4 address assigned.

Add support for dhcp-relay-disparity event which uses same repro for dhcp-relay-disparity but at greater intervals and packets sent due to dhcpmon configuration

#### How did you do it?

Added new testcase that triggered sending dhcp discover packet which will trigger event.

#### How did you verify/test it?

Pipeline/Manual test

#### Any platform specific information?

No

#### Supported testbed topology if it's a new test case?

T0

### CODE REVIEW

File: tests/telemetry/events/host_events.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 tag = "sonic-events-host"
 
 
-def test_event(duthost, gnxi_path, ptfhost, data_dir, validate_yang):
+def test_event(duthost, gnxi_path, ptfhost, ptfadapter, data_dir, validate_yang):
     logger.info("Beginning to test host events")
```
Comment (bingwang-ms) [inline tests/telemetry/events/host_events.py:16]: Is `ptfadapter` referenced in this function?

File: tests/telemetry/events/eventd_events.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 tag = "sonic-events-eventd"
 
 
-def test_event(duthost, gnxi_path, ptfhost, data_dir, validate_yang):
+def test_event(duthost, gnxi_path, ptfhost, ptfadapter, data_dir, validate_yang):
```
Comment (bingwang-ms) [inline tests/telemetry/events/eventd_events.py:11]: Is `ptfadapter` referenced in this function?

**Comment by bingwang-ms** [comment]
Can you run the test and paste the test result in PR description?

File: tests/telemetry/events/host_events.py (hunk 1 above)
Comment (zbud-msft) [reply tests/telemetry/events/host_events.py:16]: No, but test_event is a function defined in test_events.py as a function that will be called in all modules which takes in ptfadapter.

File: tests/telemetry/events/eventd_events.py (hunk 2 above)
Comment (zbud-msft) [reply tests/telemetry/events/eventd_events.py:11]: No, but test_event is a function defined in test_events.py as a function that will be called in all modules which takes in ptfadapter.

**Review by bingwang-ms — APPROVED** [review]

**Pull request merged**